### ISSUE TITLE
Segfault when inserting twice the same polyline constraint

### ISSUE THREAD

**Issue by sloriot** [CGAL/cgal#3447]
Reported on cgal-discuss.
Code: https://gist.github.com/sloriot/ea321a1020a838290ed99c98c7d8b697

The issue is in CDT+, with CDT there is no issue.

**Comment by sloriot** [comment]
The following patch fixes the segfaut:
```
diff --git a/Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h b/Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h
index 248159481e..6f13b8f7e2 100644
--- a/Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h
+++ b/Triangulation_2/include/CGAL/Constrained_triangulation_plus_2.h
@@ -582,12 +582,22 @@ private:
     CGAL_assertion(n >= 2);
     
     Constraint_id ca = hierarchy.insert_constraint(vertices[0],vertices[1]);
-    insert_subconstraint(vertices[0],vertices[1]); 
+    if (ca != NULL)
+      insert_subconstraint(vertices[0],vertices[1]); 
 
     if(n>2){
       for(std::size_t j=1; j<n-1; j++){
-	hierarchy.append_constraint(ca, vertices[j], vertices[j+1]);
-	insert_subconstraint(vertices[j], vertices[j+1]);
+        if (ca == NULL)
+        {
+          ca = hierarchy.insert_constraint(vertices[j],vertices[j+1]);
+          if (ca == NULL)
+            insert_subconstraint(vertices[j], vertices[j+1]);
+        }
+        else
+        {
+          hierarchy.append_constraint(ca, vertices[j], vertices[j+1]);
+          insert_subconstraint(vertices[j], vertices[j+1]);
+        }
       }
     }
```

However, I don't think it is sufficient. The pb here is that when we insert a constrained polyline, the first segment is used to get the context. If the segment already exist, the context is NULL and obviously not other  segments can be added.

With the fix, I'm considering the first segment of the polyline that is _new_ but it is not clear yet to me what should be done in case of overlap with two segments from different contexts.

**Comment by sloriot** [comment]
ping @afabri

**Comment by MarinLukovic** [comment]
@afabri @sloriot 
Do you have some insights what is plan for this issue? Will it be fixed in near future?
Sometimes segfault also happens when constraints are being removed from triangulation.

**Comment by sloriot** [comment]
Fixed by #3859. It will be in both 4.13.2, 4.14.1, and CGAL 5.0
